stellarphot/core.py

```diff
@@ -62,3 +70,305 @@ def __init__(self, gain=1.0 * u.electron / u.adu,
+
+                    case 'annulus_area':
+                        self.data['annulus_area'] = np.pi * (self.data['annulus_outer']**2 -self.data['annulus_inner']**2)
+
+                    case 'snr':
+                        # Since noise in counts, the proper way to compute SNR is sqrt(gain)*counts/noise
+                        self.data['snr'] = np.sqrt(self.camera.gain) * self.data['aperture_net_cnts'].value / self.data['noise'].value
+
+                    case 'bjd':
+                        self.data['bjd'] = self.add_bjd_col()
+
+                    case 'night':
+                        # Generate integer counter for nights. This should be approximately the MJD at noon local
+                        # time just before the evening of the observation.
+                        hr_offset = int(self.observatory.lon.value/15)
+                        # Compute offset to 12pm Local Time before evening
+                        LocalTime = Time(self.data['date-obs']) + hr_offset
+                        hr = LocalTime.ymdhms.hour
+                        # Compute number of hours to shift to arrive at 12 noon local time
+                        shift_hr = hr.copy()
+                        shift_hr[hr < 12] = shift_hr[hr < 12] + 12
+                        shift_hr[hr >= 12] = shift_hr[hr >= 12] - 12
+                        shift = Column(data = -shift_hr * u.hr - LocalTime.ymdhms.minute * u.min - LocalTime.ymdhms.second*u.s, name='shift')
+                        # Compute MJD at local noon before the evening of this observation
+                        self.data['night'] = Column(data=np.array((Time(self.data['date-obs']) + shift).to_value('mjd'), dtype=int), name='night')
+
+                    case 'mag_inst':
+                        self.data['mag_inst'] = -2.5 * np.log10(self.camera.gain * self.data['aperture_net_cnts'].value /
+                                                    self.data['exposure'].value)
+
+                    case 'mag_error':
+                        # data['snr'] must be computed first
+                        self.data['mag_inst'] = 1.085736205 / self.data['snr']
+
+                    case _:
+                        raise ValueError(f"Trying to compute a column ({this_col}), something that should never happen.")
+
+
+        # Apply the filter/passband name update
+        if passband_map is not None:
+            self.update_passbands()
+
+
+    def update_passbands(self):
+        # Converts filter names in filter column to AAVSO standard names
+        for orig_pb, aavso_pb in self.passband_map.items():
+            mask = self.data['passband'] == orig_pb
+            self.data['passband'][mask] = aavso_pb
+
+    def add_bjd_col(self):
+        """
+        Returns a astropy column of barycentric Julian date times corresponding to
+        the input observations.  It modifies that table in place.
+
+        Parameters
+        ----------
+
+        phot_table: `astropy.table.Table`
```

Comment:
I don't think `phot_table` is actually a parameter...

Reply:
True enough, I think I was originally thinking of passing it and realized I didn't need to.